pinot-controller/src/main/java/org/apache/pinot/controller/helix/SegmentStatusChecker.java

```diff
@@ -198,6 +217,9 @@ private void updateSegmentMetrics(String tableNameWithType, Context context) {
         // Push is not finished yet, skip the segment
         continue;
       }
+      if (segmentZKMetadata != null && segmentZKMetadata.getSizeInBytes() > 0) {
```

Comment:
(minor) Store `segmentZKMetadata.getSizeInBytes()` in a local variable to avoid reading the same field twice